File: sdk/unity/sdk/Assets/KingstonDynamics/DragonSong/Protocol/Messaging/Type/Authentication/AuthenticationLogout.cs
using KingstonDynamics.DragonSong.Protocol.Messaging.Data;
using KingstonDynamics.DragonSong.Protocol.Utils;

namespace KingstonDynamics.DragonSong.Protocol.Messaging.Type.Authentication
{
    public class AuthenticationLogout : Message
    {
        private const MessageType Type = MessageType.AUTHENTICATION_LOGOUT;

        public AuthenticationLogout(string auditId) : base((int)Type, auditId)
        {
            // Empty
        }

        public AuthenticationLogout() : this(Keyinator.GenerateGUID())
        {
            // Empty
        }

        public AuthenticationLogout(Readinator reader) : base(reader)
        {
            // Empty
        }
    }
}

File: src/protocol/src/main/java/com/kingstondynamics/dragonsong/protocol/message/type/authentication/AuthenticationLogout.java
package com.kingstondynamics.dragonsong.protocol.message.type.authentication;

import com.kingstondynamics.dragonsong.protocol.message.Message;
import com.kingstondynamics.dragonsong.protocol.message.data.MessageType;
import com.kingstondynamics.dragonsong.protocol.utils.Keyinator;

public class AuthenticationLogout extends Message {
    
    private final static MessageType type = MessageType.AUTHENTICATION_LOGOUT;
    
    public AuthenticationLogout(String auditId) {
        super(type.getType(), auditId);
    }
    
    public AuthenticationLogout() {
        this(Keyinator.generateGUID());
    }
}
